src/Rules/DoctrineKeyValueStyleRule.php

```diff
@@ -174,17 +174,19 @@ public function processNode(Node $callLike, Scope $scope): array
                 // Convert IntegerRangeType column types into IntegerType so
                 // that any integer value is accepted for integer columns,
                 // since it is uncommon to check integer value ranges.
-                if ($argColumnType instanceof IntegerRangeType) {
-                    $argColumnType = new IntegerType();
-                } elseif ($argColumnType instanceof UnionType) {
-                    $newTypes = [];
-                    foreach ($argColumnType->getTypes() as $type) {
-                        if ($type instanceof IntegerRangeType) {
-                            $type = new IntegerType();
+                if (! $valueType instanceof IntegerRangeType) {
```

Comment:
regarding such `instanceof *Type` construction I can recommend https://phpstan.org/blog/why-is-instanceof-type-wrong-and-getting-deprecated to you.

`IntegerRangeType` is not yet fully supported on the `Type`-interface.. therefore we might still need it like that though

Reply:
Agreed, I'm happy to change this once `IntegerRangeType` is supported in the new design. As you suggest, I think we would still need this as-is for now, since I don't see any methods to replace it.